### PR TITLE
document jsonb `-` and `#-` operators

### LINKED ISSUE #1439: Document: feat(expr): support jsonb `-` and `#-` operator
This issue tracks the documentation update needed for the merged PR #13118.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/13118
Source PR Merged At: 2023-10-30T10:21:42Z

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

Documented these two jsonb operators.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #1439 

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-json.md [hunk 1]
```diff
@@ -152,71 +152,84 @@ jsonb_object(array['a', null]) → {"a": null}
...
+### `jsonb - text → jsonb`
 
-Concatenates jsonb data type.
+Deletes a key (and its value) from a JSON object, or matching string value(s) from a JSON array.
 
-```bash title=Syntax
-(jsonb || jsonb) → jsonb
+```bash title=Examples
+{"a": "b", "c": "d"}'::jsonb - 'a' → {"c": "d"}
+["a", "b", "c", "b"]'::jsonb - 'b' → ["a", "c"]
```
Comment (wangrunji0408) [inline docs/sql/functions-operators/sql-function-json.md]: Missing leading `'`. same problem in the following examples.

File: docs/sql/functions-operators/sql-function-json.md [hunk 2]
```diff
@@ -232,112 +245,84 @@ SELECT '{"a": "b"}'::jsonb || '42'::jsonb;
 
 This operator checks if the left `jsonb` value contains the right `jsonb` value. For a detailed description and examples about containment and existence, see [jsonb Containment and Existence](https://www.postgresql.org/docs/current/datatype-json.html) in PostgreSQL's documentation.
 
-```sql title=Syntax
-left_jsonb_value @> right_jsonb_value → BOOLEAN
-```
-
 ```bash title=Examples
-'[1, 2, 3]'::jsonb @> '[1, 3]'::jsonb → t
+[1, 2, 3]'::jsonb @> '[1, 3]'::jsonb → t
```
Comment (wangrunji0408) [inline docs/sql/functions-operators/sql-function-json.md]: Is it an accident? seems should be reverted. 👀

File: docs/sql/functions-operators/sql-function-json.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/functions-operators/sql-function-json.md]: My bad. I mistook `'` as ```. Let me revert some of the changes.

**Review by wangrunji0408 — APPROVED** [review]
LGTM

**Pull request merged**